### DISCUSSION TITLE
How to decrease verbose informations

### DISCUSSION THREAD

**Discussion by Nyx2022** [go-resty/resty#455] [Q&A] [answered]
when the url is not reachabble,resty will print a lot of errors like:

ERROR RESTY Get "*****": read tcp ***
![image](https://user-images.githubusercontent.com/51039208/130379528-eb0475db-cfb1-4be2-a78c-e97aca924bf4.png)

i wanna know how to decrease or close these outputs

**Comment by moorereason** [comment, marked as answer]
To override the default logger, use [*Client.SetLogger](https://pkg.go.dev/github.com/go-resty/resty/v2#Client.SetLogger).

**Comment by Nyx2022** [comment]
> To override the default logger, use [*Client.SetLogger](https://pkg.go.dev/github.com/go-resty/resty/v2#Client.SetLogger).

i found the SetLogger(l Logger) *Client function,the Logger has 3 types:
type Logger interface {
	Errorf(format string, v ...interface{})
	Warnf(format string, v ...interface{})
	Debugf(format string, v ...interface{})
}

from the picture, i got a error,so , can i close the output in my picture by using resty.setlogger(Logger.Errorf) ?

**Comment by moorereason** [comment]
You need to build your own type that satisfies the Logger interface.

**Comment by Nyx2022** [comment]
> You need to build your own type that satisfies the Logger interface.

can u show me a example

**Comment by jeevatkm** [comment]
@Nyx2020 You could do something similar to what Resty does internally and override.
https://github.com/go-resty/resty/blob/b8f7c11e7ffbd132c6743f16d784ee7164a883ba/util.go#L44-L66
